microservices.yml

```diff
@@ -77,30 +98,35 @@ commands:
           command: |
             eval $(aws ecr get-login --region $AWS_DEFAULT_REGION --no-include-email)
             docker push $TAG_COMMIT
-            docker push $TAG_CI
+            docker push $TAG_BRANCH
             docker push $TAG_LATEST
+            [ ! -z $TAG_GITTAG ] && docker push $TAG_GITTAG || true
 
-  install-npm-modules-with-cache:
-    description: Installs npm modules from circleci cache (if it exists) and saves results back to cache
+  run-unit-tests-in-docker:
+    description: Runs tests in docker container
     steps:
-      - restore_cache:
-          key: rooms-npm-cache-{{ checksum "package.json" }}
       - run:
-          name: Install NPM dependencies
+          name: Run unit tests in docker container
           command: |
-            npm install
-      - save_cache:
-          key: rooms-npm-cache-{{ checksum "package.json" }}
-          paths:
-            - ./node_modules
-
-  run-docker-compose-tests:
-    description: Runs docker compose up tests with proper exit code verification to fail build
+            docker run --env LOG_TO_CONSOLE=true --env LOG_TO_FILE=true --env LOGGING_LEVEL=info --env-file common.env --env-file tables.env --env-file assa.env --env-file dorma.env -it ${TAG_COMMIT} npm run test:unit
+  run-integration-tests-in-docker:
+    description: Runs integration tests in docker container
     steps:
       - run:
-          name: Run docker compose up test with exit code flag
+          name: Run integration tests in docker container
           command: |
-            docker-compose up --exit-code-from test test
+            docker run --env LOG_TO_CONSOLE=true --env LOG_TO_FILE=true --env LOGGING_LEVEL=info --env-file common.env --env-file tables.env --env-file assa.env --env-file dorma.env -it ${TAG_COMMIT} npm run test:integration
```

Comment:
the env-file definitions are a bit questionable. We have services with many (an example is the rooms has common, logss3, logcloudwatch, snt, rover). Do we need to specify all possible options in this line?

Reply:
should be agnostic, can we source variables from `BASH_ENV` into container?

Reply:
yeah, this is my main concern... sourcing `BASH_ENV` is hacky at least...

my proposition is to use `Makefile` in each repo and run `make ci:test:unit`/`make ci:test:integration`. that way we can do lots of stuff the same in each repo but also we can have some exception per repo and still use the common way of running on Circle, also we can "source" `BASH_ENV` inside `Makefile`

what do you think guys @codin-sf @bartek-sf @afaq-sf @tomasz-sf @patrick-sf

Reply:
I like this idea

👍